### PR TITLE
Add 1.4 release information

### PR DESCRIPTION
**Pull request by annajung**
Signed-off-by: Anna Jung (VMware) <antheaj@vmware.com>

**Description of your changes:**
Based on [discussion of improving file structure for the release](https://kubeflow.slack.com/archives/C9V2WT2KV/p1628727885011700),
- rename directory `release` to `releases`
- add `release-1.4` directory
- add `release-1.4/README.md` with release timeline and relevant links
- add `release-1.4/release-team.md` with name of all release members

**Checklist:**
- [x] Unit tests pass:
  **Make sure you have installed kustomize == 3.2.1**
    1. `make generate-changed-only`
    2. `make test`

cc @RFMVasconcelos @DavidSpek @mkbhanda @kimwnasptd

### CODE REVIEW

File: docs/releases/release-1.4/release-team.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Kubeflow 1.4 Release Team
+
+| **Role** | **Name** (**GitHub / Slack ID**) |
+|----------|----------------------------------|
+| Release Manager | Kimonas Sotirchos (GitHub: [@kimwnasptd](https://github.com/kimwnasptd) / Slack: `@kimwnasptd`) |
+| Release Team Member(s) | David van der Spek (GitHub: [@DavidSpek](https://github.com/DavidSpek) / Slack: `@David van der Spek`), Malini Bhandaru (GitHub: [@mkbhanda](https://github.com/mkbhanda) / Slack: `@Malini Bhandaru`), and Rui Vasconcelos (GitHub: [@RFMVasconcelos](https://github.com/RFMVasconcelos) / Slack: `@Rui Vasconcelos`) |
+| Product Manager |  |
+| Docs Lead | Rui Vasconcelos (GitHub: [@RFMVasconcelos](https://github.com/RFMVasconcelos) / Slack: `@Rui Vasconcelos`) |
+| Working Group Liaison(s) |  |
+| Distribution Representative |  |
```
Comment (annajung) [inline docs/releases/release-1.4/release-team.md:10]: Do we have anyone serving as a product manager? WG liaison? distribution rep?

File: docs/releases/release-1.4/release-team.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+| Shadow(s) | Anna Jung (GitHub: [@annajung](https://github.com/annajung) / Slack: `@annajung`) |
```
Comment (annajung) [inline docs/releases/release-1.4/release-team.md:11]: I believe there was one more person interested in being a shadow. @RFMVasconcelos could you help gather the info needed to be added here?

**Comment by RFMVasconcelos** [comment]
Thank you for leading this @annajung! 

I know that @knkski - Kenneth Koski - wanted to join as a shadow. Can you please confirm @knkski?

If that's the case, @annajung could you please add @knkski to this PR?

File: docs/releases/release-1.4/release-team.md (hunk 1 above)
Comment (annajung) [reply docs/releases/release-1.4/release-team.md:10]: hi @jbottum, during the release team meeting today, your name was brought up as the product manager for the 1.4 release. Could you confirm your commitment to the role? Once confirmed, I'll update the markdown to add your info.

**Comment by knkski** [comment]
Yes, I would be interested in shadowing for this release :+1:

File: docs/releases/release-1.4/release-team.md (hunk 1 above)
Comment (kimwnasptd) [reply docs/releases/release-1.4/release-team.md:10]: Regarding the WG liaisons I'd propose the following people for this release, who have helped with the [release issues](https://github.com/orgs/kubeflow/projects/46):

| WG | handle |
| --- | --- |
| AutoML | @andreyvelich |
| Manifests | @kimwnasptd |
| Notebooks | @kimwnasptd |
| Pipelines | @Bobgy @zijianjoy |
| Serving | @pvaneck |
| Training | @johnugeorge @Jeffwan @terrytangyuan |

@annajung @mkbhanda we could also add a bullet for defining the WG liaisons in the [preparation phase](https://github.com/kubeflow/manifests/blob/master/docs/release/handbook.md#preparation). WDYT?

Comment (jbottum) [reply docs/releases/release-1.4/release-team.md:10]: @annajung yes, I will act as the product manager for the 1.4 release.

Comment (annajung) [reply docs/releases/release-1.4/release-team.md]: member list updated with the proposed WG liaisons and product manager, PTAL

Comment (annajung) [reply docs/releases/release-1.4/release-team.md]: > we could also add a bullet for defining the WG liaisons in the preparation phase. WDYT?

@kimwnasptd  Do you mean add a step to find out who will be acting as a liaison for each WG during preparation phase?

Comment (kimwnasptd) [reply docs/releases/release-1.4/release-team.md]: > @kimwnasptd Do you mean add a step to find out who will be acting as a liaison for each WG during preparation phase?

Correct. Essentially the release team will need to know who these people are early on so they can start pinging them in issues, i.e. 2 weeks before the feature freeze to provide them branches/versions etc.

**Comment by Bobgy** [comment]
/lgtm
thanks for the early update!
Feel free to ping me when the PR is ready

File: docs/releases/release-1.4/release-team.md (hunk 1 above)
Comment (annajung) [reply docs/releases/release-1.4/release-team.md]: done!

Comment (kimwnasptd) [reply docs/releases/release-1.4/release-team.md]: Thanks!

**Comment by kimwnasptd** [comment]
Thank you for pushing this @annajung! Let's merge this first iteration to have the schedule and team committed to GH.
I'll follow up with an issue to get the distribution reps for this release, so that we can add them in a subsequent PR.

/lgtm
/approve

**Pull request merged**